tools/globalids.py

```diff
@@ -0,0 +1,212 @@
+import numpy as np
+import torch
+import operator
+import itertools
+
+from gloo_wrapper import allgather_sizes, alltoall_cpu, alltoallv_cpu
+
+def get_global_node_ids(rank, world_size, nodeids_ranks, partitions, node_data):
+    """
+    For non-local node, whose orig-node-id <-> global-id is not present at the current rank
+    retrieve their global_ids. 
+
+    Parameters: 
+    -----------
+    rank : rank of the process
+    world_size : total no. of ranks configured
+    nodeids_ranks : list of lists of orig-node-ids, index of the list is the rank of the process
+                    where orig-node-id <-> global_id mapping is located. 
```

Comment:
first, `nodeids_ranks` cannot be a list of lists. i assume the total number of elements in the list of lists is the number of nodes in a graph. we should use a list of numpy arrays or pytorch arrays to store it.

i don't quite understand the mapping. does it map the global IDs to original node IDs? the index is global IDs? what is the original node IDs? the node IDs within a node type? or something else?